coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveToRelConverter.java

```diff
@@ -92,12 +93,16 @@ protected SqlToRelConverter getSqlToRelConverter() {
 
   @Override
   protected SqlNode toSqlNode(String sql, Table hiveView) {
-    return parseTreeBuilder.process(trimParenthesis(sql), hiveView);
+    final SqlNode sqlNode = parseTreeBuilder.process(trimParenthesis(sql), hiveView);
+    if (hiveView != null) {
+      sqlNode.accept(new FuzzyUnionSqlRewriter(hiveView.getTableName(), this));
+    }
+    return sqlNode.accept(new HiveToCoralOperatorConverter(getSqlValidator()));
```

Comment:
the sqlValidator has not validated the node yet. The validation process populates the SqlValidator with the context of the RelNode here https://github.com/linkedin/coral/blob/master/coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveSqlToRelConverter.java#L59 . Before this step, the sqlValidator is empty.

If we are going to use an empty sqlValidator and populate it inside each "OperatorTransformer", then why do we pass it at all? Should we consider manufacturing it inside HiveToCoralOperatorConverter or OperatorTransformer?

Reply:
In order to create a new SqlValidator in `OperatorTransformer`/`HiveSqlNodeToCoralSqlNodeConverter`, we need to provide/instantiate lots of required parameters, because [the instantiation of sqlValidator is pretty complex](https://github.com/linkedin/coral/blob/39d4ba362cf4cd2fd83874ed2ef2f607225b328f/coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/HiveToRelConverter.java#L53). 
Therefore, I would prefer to use the one in `HiveToRelConverter` directly for simplification, and it doesn't cause regression.